crates/oxc_linter/src/rules/deepscan/uninvoked_array_callback.rs

```diff
@@ -0,0 +1,96 @@
+use oxc_ast::{
+    ast::{Argument, Expression, MemberExpression},
+    AstKind, GetSpan, Span,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("Uninvoked array callback")]
+#[diagnostic(
+    severity(warning),
+    help("consider filling the array with `undefined` values using `Array.prototype.fill()`")
+)]
+struct UninvokedArrayCallbackDiagnostic(
+    #[label("this callback function will not be invoked")] Span,
+    #[label("because this is an array with only empty slots")] Span,
+);
+
+/// `https://deepscan.io/docs/rules/uninvoked-array-callback`
+#[derive(Debug, Default, Clone)]
+pub struct UninvokedArrayCallback;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// This rule applies when the Array functions having callback argument are used for an array with empty slots.
+    ///
+    /// ### Why is this bad?
+    /// When the Array constructor is called with a single number argument, an array with the specified number of empty slots (not actual undefined values) is constructed.
+    /// If a callback function is passed to the function of this array, the callback function is never invoked because the array has no actual elements.
+    ///
+    ///
+    /// ### Example
+    /// ```javascript
+    ///   const list = new Array(5).map(_ => createElement());
+    /// ```
+    UninvokedArrayCallback
+);
+
+impl Rule for UninvokedArrayCallback {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let new_expr = if let AstKind::NewExpression(new_expr) = node.get().kind()
+            && new_expr.callee.is_specific_id("Array")
+            && new_expr.arguments.len() == 1
+            && let Some(Argument::Expression(arg_expr)) = new_expr.arguments.iter().next()
```

Comment:
Didn't know you could chain `let` in a condition like this 🔥

Reply:
same 👌

Reply:
We are on nightly, this is let-chains